### ISSUE TITLE
Remove arbitrary 180 degree rotation of target pose

### ISSUE THREAD

**Issue by marip8** [ros-industrial/reach#77]
The reach study currently implicitly rotates each target pose by 180 degrees about the x-axis and makes the assumption that a robot's TCP frame has the z-axis pointing "out". This assumption and implict operation can be easily missed, causing reach studies to report many infeasible IK solutions, which is generally difficult to debug. We should remove the implicit rotation and state the assumption that the reach study will attempt to align the TCP frame directly with the target poses (per the operation of the underlying IK solver).

https://github.com/ros-industrial/reach/blob/d89ddafadcf87350dfd670b5113a3e4c1577a353/src/reach_study.cpp#L111

**Comment by nikhil-sethi** [comment]
When this rotation is removed, the IK solver does not give any solutions though.

I'm able to get the same set of solutions (with the targets flipped) when I flip the `tgt_frame` before passing to `evauateIK`. Does this mean that the IK solver itself expects opposing z axes?

```cpp
std::tie(solution, score) = evaluateIK(tgt_frame* Eigen::AngleAxisd(M_PI, Eigen::Vector3d::UnitX()), params_.seed_state, ik_solver_, evaluator_);
```

<img width="544" height="641" alt="Image" src="https://github.com/user-attachments/assets/fcef5d81-4b34-4fb6-8ce8-01d46d78f04b" />

**Comment by marip8** [comment]
> When this rotation is removed, the IK solver does not give any solutions though.

You won't be able to just remove this rotation alone. You'll also need to rotate the TCP frame 180 degrees in your URDF to compensate.

The idea here is that the visualization should show the TCP frame aligning directly with the visualized poses, which is what most people assume when they think about inverse kinematics

**Comment by nikhil-sethi** [comment]
Yep I agree with the idea and also updated the urdf but the observations are a bit different.

When I remove the implicit rotation (z axis of target frames inside the hood), **and** update the URDF (z-axis of tcp inward), I do get solutions. But the visualizations show that they are not aligned. They are opposing.
All this is without the hack that i mentioned in the previous comment

<img width="230" height="422" alt="Image" src="https://github.com/user-attachments/assets/739a88df-3a8d-49e9-8f25-ba0d3400e5b0" />

Basically, I can only get solutions when the frames are opposing each other. But it's likely that I'm messing up the logic somewhere. Let me know!

**Comment by marip8** [comment]
Looks like there is a [corresponding z-axis flip in the visualization of the markers in the `reach_ros2` package](https://github.com/ros-industrial/reach_ros2/blob/769735195bc7075756f0745389d54be4e620257b/src/utils.cpp#L125) that needs to be removed as well

**Comment by nikhil-sethi** [comment]
Hey @marip8, I've added https://github.com/ros-industrial/reach/pull/90 and https://github.com/ros-industrial/reach_ros2/pull/46 to fix this.
